src/components/cards/ProductCard.tsx

```diff
@@ -3,68 +3,90 @@ import { FaEye } from "react-icons/fa";
 import { CiHeart } from "react-icons/ci";
 import React, { useState } from "react";
 import { Link } from "react-router-dom";
+import { ToastContainer, toast } from "react-toastify";
+import { AxiosError } from "axios";
 
 import { IProduct } from "../../types";
+import { useAppDispatch } from "../../redux/hooks";
+import { addToCart } from "../../redux/reducers/cartSlice";
+import Warning from "../common/notify/Warning";
 
 interface IProductCardProps {
   product: IProduct;
 }
 
 const ProductCard: React.FC<IProductCardProps> = ({ product }) => {
   const [isHovered, setIsHovered] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const dispatch = useAppDispatch();
 
   const soleilFN = (price: number) => {
     if (price < 1000) {
       return price.toString();
-    } if (price >= 1000 && price < 1000000) {
+    }
+    if (price >= 1000 && price < 1000000) {
       return `${(price / 1000).toFixed(1)}k`;
     }
     return `${(price / 1000000).toFixed(1)}M`;
   };
 
+  const handleAddToCart = async () => {
+    if (!localStorage.getItem("accessToken")) {
+      toast.info("Please Log in to add to cart.");
+      return;
+    }
+    setIsLoading(true);
+    try {
+      await dispatch(
+        addToCart({ productId: product.id, quantity: 1 }),
+      ).unwrap();
+    } catch (err) {
+      const error = err as AxiosError;
+      toast.error(`Failed to add product to cart: ${error.message}`);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
   const name = product.name.length > 20
     ? `${product.name.substring(0, 12)}...`
     : product.name;
 
   const date = new Date(product.createdAt).getDate();
-  const currrentDate = new Date().getDate();
-
-  const diff = currrentDate - date;
+  const currentDate = new Date().getDate();
 
+  const diff = currentDate - date;
   return (
-    <div
-      className=" max-h-[270px] bg-[#F5F5F5] mb-2 relative"
-      data-testid="tbt"
-    >
+    <div className="max-h-[270px] bg-[#F5F5F5] mb-2 relative" data-testid="tbt">
       {diff < 2 && (
-        <div className="absolute top-2 left-2 " data-testid="new">
-          <p className=" rounded-md z-40  text-white new-product ">New</p>
+        <div className="absolute top-2 left-2" data-testid="new">
+          <p className="rounded-md z-40 text-white new-product">New</p>
         </div>
       )}
-
+      <ToastContainer />
       <div
-        className=" relative min-h-[200px]"
+        className="relative min-h-[200px]"
         onMouseOver={() => setIsHovered(true)}
         onMouseLeave={() => setIsHovered(false)}
       >
         <img
           src={product.images[0]}
-          alt="d-ss"
+          alt="product"
           className="w-[100%] h-[200px] hover:scale-[1.05] transition duration-200"
           data-testid="prod-image"
         />
         {isHovered && (
           <button
```

Comment:
From here u have to conditianla render add to cart btn, if product is added in cart u don't need to show add to cart button again instead u have to render  remove from cart button 

https://github.com/atlp-rwanda/eagles-ec-fe/assets/105858896/c74612e6-88e1-4b8f-acb8-2a7e9ae8e97b